Common/Systems/ModPlayers/SkilPassivePlayer.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using PathOfTerraria.Common.Mechanics;
+using Terraria.ModLoader.IO;
+
+namespace PathOfTerraria.Common.Systems.ModPlayers;
+
+public class SkillPassivePlayer : ModPlayer
+{
+	// A dictionary to store the acquired passive points for each skill.
+	public Dictionary<Skill, int> AcquiredPassivePoints = new();
```

Comment:
Skills are singletons, correct? If so, this is fine. But especially if they aren't, I'd use the name of the skill as the key instead of the skill itself for instancing reasons, and names are easier to save, load, send and receive.